File: Backend/src/test/java/lan/dk/podcastserver/scheduled/ItemDeletionScheduledTest.java
package lan.dk.podcastserver.scheduled;

import lan.dk.podcastserver.business.update.UpdatePodcastBusiness;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.runners.MockitoJUnitRunner;

import static org.mockito.Mockito.times;
import static org.mockito.Mockito.verify;

/**
 * Created by kevin on 17/08/15 for Podcast Server
 */
@RunWith(MockitoJUnitRunner.class)
public class ItemDeletionScheduledTest {

    @Mock UpdatePodcastBusiness updatePodcastBusiness;
    @InjectMocks ItemDeletionScheduled itemDeletionScheduled;

    @Test
    public void should_delete_old_item() {
        /* When */ itemDeletionScheduled.deleteOldItem();
        /* Then */ verify(updatePodcastBusiness, times(1)).deleteOldEpisode();
    }

    @Test
    public void should_delete_old_cover() {
        /* When */ itemDeletionScheduled.deleteOldCover();
        /* Then */ verify(updatePodcastBusiness, times(1)).deleteOldCover();
    }

}

File: backend/src/test/kotlin/com/github/davinkevin/podcastserver/scheduled/ItemDeletionScheduledTest.kt
package com.github.davinkevin.podcastserver.scheduled

import com.github.davinkevin.podcastserver.business.update.UpdatePodcastBusiness
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.mockito.InjectMocks
import org.mockito.Mock
import org.mockito.Mockito.times
import org.mockito.Mockito.verify
import org.mockito.junit.jupiter.MockitoExtension

/**
 * Created by kevin on 17/08/15 for Podcast Server
 */
@ExtendWith(MockitoExtension::class)
class ItemDeletionScheduledTest {

    @Mock lateinit var updatePodcastBusiness: UpdatePodcastBusiness
    @InjectMocks lateinit var itemDeletionScheduled: ItemDeletionScheduled

    @Test
    fun `should delete old item`() {
        /* When */ itemDeletionScheduled.deleteOldItem()
        /* Then */ verify<UpdatePodcastBusiness>(updatePodcastBusiness, times(1)).deleteOldEpisode()
    }

    @Test
    fun `should delete old cover`() {
        /* When */ itemDeletionScheduled.deleteOldCover()
        /* Then */ verify<UpdatePodcastBusiness>(updatePodcastBusiness, times(1)).deleteOldCover()
    }

}
